### PR TITLE
Upgrade az vmware extension to use 2023-03-01 API version

### PR DESCRIPTION
**Pull request by jonathanhe-msft**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by jonathanhe-msft** [comment]
@cataggar I saw that provisioned_networks list and get commands were added to the swagger for 2023-03-01 API. I didn't implement these commands in this PR. If they need to be implemented before this is merged please let me know and I will update.

I didn't make any changes for the cloud error changes. The only changes to CLI are new functionality for the extended network blocks.

File: src/vmware/HISTORY.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Release History
 
+## 6.0.1 (2023-09)
+- Update to AVS 2023-03-01 API
+- Add `az vmware private-cloud add-extended-network-blocks`
+- Add `az vmware private-cloud delete-extended-network-blocks`
+- Updated `az vmware private-cloud create` with new `ext_nw_blocks` parameter
```
Comment (cataggar) [inline src/vmware/HISTORY.md:7]: Is spelling it out with `extended_network_blocks` to long? Does something not work?

Comment (jonathanhe-msft) [reply src/vmware/HISTORY.md:7]: Yea there's a linter error that parameter name is too long

**Review by cataggar — APPROVED** [review]

**Comment by jonathanhe-msft** [comment]
@jsntcy can you please review?

File: src/vmware/setup.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 from io import open
 from setuptools import setup, find_packages
 
-VERSION = "5.0.3"
+VERSION = "6.0.0"
```
Comment (zhoxing-ms) [inline src/vmware/setup.py:11]:
```suggestion
VERSION = "5.1.0"
```
In fact, we only recommend upgrading the major version when the PR changes include breaking changes. If only some new features are released, it is recommended to upgrade the secondary version.

**Comment by cataggar** [comment]
> @cataggar I saw that provisioned_networks list and get commands were added to the swagger for 2023-03-01 API. I didn't implement these commands in this PR. If they need to be implemented before this is merged please let me know and I will update.
> 
> I didn't make any changes for the cloud error changes. The only changes to CLI are new functionality for the extended network blocks.

The error handling shouldn't change. Correct, the only changes for this API is the extended network blocks.

**Comment by cataggar** [comment]
Replaced by #6765

**Pull request closed without merging**